src/h5web/providers/hsds/models.ts

```diff
@@ -38,11 +39,7 @@ export interface HsdsExternalLink extends Omit<HDF5ExternalLink, 'file'> {
   h5domain: string;
 }
 
-export type HsdsLink =
-  | HDF5RootLink
-  | HDF5HardLink
-  | HDF5SoftLink
-  | HsdsExternalLink;
+export type HsdsLink = HDF5HardLink | HDF5SoftLink | HsdsExternalLink;
```

Comment:
Could `HDF5ExternalLink` have the same signature as `HsdsExternalLink` ?

This way, there will no longer be a distinction between `HsdsLink` and `HDF5Link`.

Reply:
Not sure how. One has a `file` key; the other has an `h5domain` key. 🤔

Reply:
Yes but is the model with the `file` key needed elsewhere ? or could we use the `h5domain` key all the way ?

Reply:
Well, just from the name, `h5domain` seems very specific to HSDS. The HDF5 spec uses the term "file": http://davis.lbl.gov/Manuals/HDF5-1.8.7/UG/09_Groups.html

Reply:
Ok nevermind, let's keep things as they are then